strings/is_substring_palindrome.py

```diff
@@ -0,0 +1,61 @@
+MAX_LEN = 1000
+
+text = ""
+dp = [[False for i in range(MAX_LEN)] for i in range(MAX_LEN)]
+
+def preprocess(s):
+    """
+        Preprocesses a string using dynamic programming.
+        Time complexity: O(n^2)
+    """
+    global text

```

Comment:
don't use global vavriables here. It'll be much better if you return the matrix (`dp`) as a result, the `text` variable is useless at all.

```py
def process(s):
    ...
    n = len(s)
    matrix = [[False for i in range(n)] for i in range(n)]
    ...
    
    return matrix

...
if __name__ == '__main__':
    ...
    matrix = process(s)
    ...
    if is_polindrome(matrix, l, r):
        ...
```

So you can delete the globals `MAX_LEN`, `text` and `dp`